content/doc/developer/publishing/documentation.adoc

```diff
@@ -0,0 +1,169 @@
+---
+title: Plugin Documentation
+layout: developersection
+---
+
+Documentation is an important part of any Jenkins plugin.
+It includes user documentation (plugin pages, changelogs, user guidelines, etc.)
+and the contributor documentation (how to contribute, developer guidelines, etc.).
+This section provides overview of these documentation types.
+
+== Plugin pages
+
+Plugin pages are hosted on the link:https://plugins.jenkins.io/[Plugin Site],
+these pages are being generated automatically using the metadata from the latest plugin release and an external documentation page.
+External documentation can be retrieved from GitHub or from https://wiki.jenkins.io[Jenkins Wiki].
+Since the introduction of GitHub documentation support in September 2019
+(https://groups.google.com/forum/#!topic/jenkinsci-dev/VSdfVMDIW-A[announcement]),
+this is the *recommended way* to host documentation.
+
+In Jenkins we recommend storing documentation in the plugin repository.
+It allows plugin maintainers to provide the same documentation from GitHub README pages and the Jenkins plugin site,
+and at the same time it allows using the Documentation-as-Code techniques when the documentation is a part of the
+repository and hence all common practices can be applied 
+(versioning, pull requests and reviews, creating documentation in parallel with features, editing docs from GitHub Web UI).
```

Comment:
FWIW this looks like a list that deserves to be formatted as a list and extended a bit, as you probably want to highlight the benefits of this approach.